airbyte-integrations/connectors/destination-clickhouse-v2/src/main/kotlin/io/airbyte/integrations/destination/clickhouse_v2/client/ClickhouseSqlGenerator.kt

```diff
@@ -380,9 +383,15 @@ fun AirbyteType.toDialectType(): String =
         TimestampTypeWithoutTimezone -> DATETIME_WITH_PRECISION
         is ArrayType,
         ArrayTypeWithoutSchema,
-        is ObjectType,
+        is UnionType,
```

Comment:
wondering: why not make _all_ of these into JSON? that's what we're doing in bigquery, and iirc mssql+s3datalake

in particular, IMO object with empty/without schema should behave identically to Object, for warehouse destinations

Reply:
checking - does the JSON type require actual JSON _object_? or does it accept any JSON _value_?

Reply:
e.g. with bigquery's json type, these are all valid:
```sql
select
  JSON'{"foo": "bar"}', -- object
  JSON'"arst"', -- string
  JSON'true', -- boolean
  JSON'42' -- int
```

Reply:
It has to be a JSON value https://clickhouse.com/docs/sql-reference/data-types/newjson